website/blog/2022/01/21/apisix-hashicorp-vault-integration.md

```diff
@@ -0,0 +1,373 @@
+
+2. RS256 RSA keypairs, both public and private keys are stored in the vault.
+
+   ```shell
+   $ curl http://127.0.0.1:9080/apisix/admin/consumers -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+    {
+        "username": "jim",
+        "plugins": {
+            "jwt-auth": {
+                "key": "rsa-keypair",
+                "algorithm": "RS256",
+                "vault": {}
+            }
+        }
+    }'
+   ```
+
+   The plugin looks up for `public_key` and `private_key` keys inside vault kv path (`<vault.prefix from conf.yaml>/consumer/jim/jwt-auth`) for `jim` mentioned inside plugin vault configuration. If not found, authentication fails.
+
+   If you are unsure, how to store public and private keys into vault kv storage, use this command
+
+   ```shell
+   # provided, your current directory contains the files named "public.pem" and "private.pem"
+    $ vault kv put kv/apisix/consumer/jim/jwt-auth public_key=@public.pem private_key=@private.pem
+    Success! Data written to: kv/apisix/consumer/jim/jwt-auth
+   ```
+
+3. Public key in consumer configuration, while the private key is in the vault.
+
+   ```shell
+   $ curl http://127.0.0.1:9080/apisix/admin/consumers -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+    {
+        "username": "john",
+        "plugins": {
+            "jwt-auth": {
+                "key": "user-key",
+                "algorithm": "RS256",
+                "public_key": "-----BEGIN PUBLIC KEY-----\n……\n-----END PUBLIC KEY-----"
+                "vault": {}
+            }
+        }
+    }'
+   ```
+
+   This plugin uses RSA public key from consumer configuration and uses the private key directly fetched from the vault.
+
+### Disable Vault from Plugin
+
+Now, to disable the vault lookup from the `jwt-auth` plugin simply remove the empty vault object from the consumer plugin configuration (in this case it is `jack`). This will make the jwt plugin to lookup signing secrets (both HS256/HS512 or RS512 keypairs) into plugin configuration for subsequent requests to the URI route where the `jwt-auth` configuration has been enabled. Even if you have vault configuration enabled in APISIX `config.yaml` no request will be sent to the vault server.
+
+APISIX plugins are hot-reloaded, therefore is no need to restart APISIX.
+
+```shell
+$ curl http://127.0.0.1:9080/apisix/admin/consumers -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
+{
+    "username": "jack",
+    "plugins": {
+        "jwt-auth": {
+            "key": "test-key",
+            "secret": "my-secret-key"
+        }
+    }
+}'
+```
+
+## Summary
+
+This article brings you the upcoming release of the Vault-Apache APISIX integration and related details.
+
+Feel free to start a discussion in [GitHub Discussions](https://github.com/apache/apisix/discussions) or communicate via the [mailing list](https://apisix.apache.org/zh/docs/general/subscribe-guide).
```

Comment:
```suggestion
Feel free to start a discussion in [GitHub Discussions](https://github.com/apache/apisix/discussions) or communicate via the [mailing list](https://apisix.apache.org/docs/general/subscribe-guide).
```

Btw, just wondering if it is possible to add a ref to my blog? Hehe😅

Reply:
sure